### ISSUE TITLE
Red Nodes of Mirth

### ISSUE THREAD

**Issue by YanYas** [microdee/IronHydra#1]
Hi MD,

I ran your perlin demo. with the vobjects from the vvvv website, but the primitive object nodes were red and the error messsage was as follows:

> Cannot perform runtime binding on a null reference

  > at CallSite.Target(Closure , CallSite , Object )
  > at Microsoft.Scripting.Utils.ReflectionUtils.GetMethod(Dele>gate d)
  > at IronPython.Runtime.PythonDynamicStackFrame.GetMethod(CodeContext context, FunctionCode funcCode)
  > at IronPython.Runtime.PythonDynamicStackFrame..ctor(CodeContext context, FunctionCode funcCode, Int32 line)
  > at IronPython.Runtime.Ope>rations.PythonOps.Up>dateStackTrace(Exception e, CodeContext context, FunctionCode funcCode, Int32 line)
  > at execute$1(Closure , PythonFunction )
  > at <unnamed>(Closure , CodeContext , FunctionCode )
  > at IronPython.Compiler.PythonScriptCode.RunWorker(CodeContext ctx)
  > at Microsoft.Scripting.Hosting.CompiledCode.Execute[T](ScriptScope scope)
  > at VVVV.IronHydra.PythonThread.<Run>b__59_0() in C:\vvvv\libs\mcropack\packs\IronHydra\src\IronHydra\PythonThread.cs:line 108
  > at CallSite.Target(Closure , CallSite , Object )
  > at Microsoft.Scripting.Utils.ReflectionUtils.GetMethod(Dele>gate d)
  > at IronPython.Runtime.PythonDynamicStackFrame.GetMethod(CodeContext context, FunctionCode funcCode)
  > at IronPython.Runtime.PythonDynamicStackFrame..ctor(CodeContext context, FunctionCode funcCode, Int32 line)
  > at IronPython.Runtime.Ope>rations.PythonOps.Up>dateStackTrace(Exception e, CodeContext context, FunctionCode funcCode, Int32 line)
  > at execute$1(Closure , PythonFunction )
  > at <unnamed>(Closure , CodeContext , FunctionCode )
  > at IronPython.Compiler.PythonScriptCode.RunWorker(CodeContext ctx)
  > at Microsoft.Scripting.Hosting.CompiledCode.Execute[T](ScriptScope scope)
  > at VVVV.IronHydra.PythonThread.<Run>b__59_0() in C:\vvvv\libs\mcropack\packs\IronHydra\src\IronHydra\PythonThread.cs:line 108

I don't think the problem is the iron hydra nodes but I tested with a couple different vobject packages and got even worse results, like total link breakages and that sort of thing.

**Comment by microdee** [comment]
yeah i need to check that, are you using vobjects from github?

2016-01-09 16:51 GMT+00:00 YanYas <notifications@github.com>:

**Comment by microdee** [comment]
also note that IronHydra is still kindof in a proof-of-concept phase, for instance that perlin example is not finished yet and I can 85% guarantee it won't work now :(

**Comment by YanYas** [comment]
No probs, you showed a different example where you had what looked like a more vanilla python script running, is that testable?

**Comment by microdee** [comment]
absolutely, only the perlin example is not ready the other's are fine and they should provide enough kickstart to implement your own python code. I might have an idea on what is causing the exception: you have to construct the primitiveobjects you'd like to use before binding it for python

**Comment by YanYas** [comment]
Cheers Dave, 
I couldn't quite follow what was happening in the callmenames patch but I'll take a look at it. Out of curiosity, is ironpython basically asynchronous too? 
Hayd

**Comment by microdee** [comment]
IronPython is not fundamentally asynchronous but nothing preventing you putting it in a .net task (like how it's done in IronHydra) ;)
Also IronPython is not only about running python in your C# project, it can also import any .net library, basically making python a .net framework compatible language

**Comment by YanYas** [comment]
Cool, 
I'm working with opencv on python at the mo so I'll see if I can tie things together.
